samples/python/multiview_calibration.py

```diff
@@ -0,0 +1,333 @@
+        plotProjection(image_points[cam_idx][frame_idx], pattern_points, rvecs0[frame_idx],
+            tvecs0[frame_idx], rvecs[cam_idx], Ts[cam_idx], Ks[cam_idx], distortions[cam_idx],
+            is_fisheye[cam_idx], cam_idx, frame_idx, 1e2*(errors_per_frame[cam_idx,frame_idx]<errors).sum()/len(errors), image)
+    plot(detection_mask_idxs[0,0], detection_mask_idxs[1,0])
+    plt.show()
+
+def chessboard_points(grid_size, dist_m):
+    pattern = np.zeros((grid_size[0]*grid_size[1],3), np.float32)
+    pattern[:,:2] = np.mgrid[0:grid_size[0],0:grid_size[1]].T.reshape(-1,2)*dist_m # only for (x,y,z=0)
+    return pattern
+
+def circles_grid_points(grid_size, dist_m):
+    pattern = []
+    for i in range(grid_size[0]):
+        for j in range(grid_size[1]):
+            pattern.append([j*dist_m, i*dist_m, 0])
+    return np.array(pattern, dtype=np.float32)
+
+def asym_circles_grid_points(grid_size, dist_m):
+    pattern = []
+    for i in range(grid_size[1]):
+        for j in range(grid_size[0]):
+            if i % 2 == 1:
+                pattern.append([(j+.5)*dist_m, dist_m*(i//2+.5), 0])
+            else:
+                pattern.append([j*dist_m, (i//2)*dist_m, 0])
+    return np.array(pattern, dtype=np.float32)
+
+def detect(cam_idx, frame_idx, img_name, pattern_type, grid_size, criteria, winsize, RESIZE_IMAGE):
+    print(img_name)
+    assert os.path.exists(img_name)
+    img = cv.cvtColor(cv.imread(img_name), cv.COLOR_BGR2GRAY)
+    img_size = img.shape[:2][::-1]
+
+    scale = 1.0
+    img_detection = img
+    if RESIZE_IMAGE:
+        scale = 1000.0 / max(img.shape[0], img.shape[1])
+        if scale < 1.0:
+            img_detection = cv.resize(img, (int(scale * img.shape[1]), int(scale * img.shape[0])), interpolation=cv.INTER_AREA)
+
+    if pattern_type.lower() == 'checkerboard':
+        ret, corners = cv.findChessboardCorners(img_detection, grid_size, None)
+    elif pattern_type.lower() == 'circles':
+        ret, corners = cv.findCirclesGrid(img_detection, patternSize=grid_size, flags=cv.CALIB_CB_SYMMETRIC_GRID)
+    elif pattern_type.lower() == 'acircles':
+        ret, corners = cv.findCirclesGrid(img_detection, patternSize=grid_size, flags=cv.CALIB_CB_ASYMMETRIC_GRID)
+    else:
+        raise "Calibration pattern is not supported!"
+
+    if ret:
+        if scale < 1.0:
+            corners /= scale
+        corners2 = cv.cornerSubPix(img, corners, winsize, (-1,-1), criteria)
+        # cv.drawChessboardCorners(img, grid_size, corners2, ret)
+        # plt.imshow(img)
+        # plt.show()
+        return cam_idx, frame_idx, img_size, np.array(corners2, dtype=np.float32).reshape(-1,2)
+    else:
+        return cam_idx, frame_idx, img_size, np.array([], dtype=np.float32)
+
+def calibrateFromImages(files_with_images, grid_size, pattern_type, is_fisheye, dist_m, winsize, points_json_file, debug_corners, RESIZE_IMAGE=True, find_intrinsics_in_advance=False, is_parallel_detection=True):
```

Comment:
Sometimes resizing image is very convenient, if resolution is too big, it could take minutes to detect corners. While approximate corners could be found on the resized image and afterwards refined with cv::cornerSubPix.
I added argument flag, the default parameter is now False (no resize).